sdk/identity/azure-identity/src/main/java/com/azure/identity/implementation/IdentityClient.java

```diff
@@ -137,28 +139,23 @@
     public Mono<AccessToken> authenticateWithPfxCertificate(String pfxCertificatePath, String pfxCertificatePassword,
                                                             TokenRequestContext request) {
         String authorityUrl = options.getAuthorityHost().replaceAll("/+$", "") + "/" + tenantId;
-        try {
+        return Mono.fromCallable(() -> {
             ConfidentialClientApplication.Builder applicationBuilder =
-                ConfidentialClientApplication.builder(clientId,
-                    ClientCredentialFactory.create(new FileInputStream(pfxCertificatePath), pfxCertificatePassword))
-                    .authority(authorityUrl);
+                    ConfidentialClientApplication.builder(clientId, ClientCredentialFactory.createFromCertificate(
+                                new FileInputStream(pfxCertificatePath), pfxCertificatePassword))
+                            .httpClient(new HttpPipelineAdapter(options.getHttpPipeline()))
```

Comment:
so, here we don't check if (options.getHttpPipeline() != null)  before setting the pipeline ?

Reply:
Removing in favor of the other checked one